lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -2721,30 +2721,51 @@ _SPIRV_OP(PowN, true, 10)
 
 class SPIRVAtomicInstBase : public SPIRVInstTemplateBase {
 public:
+  llvm::Optional<ExtensionID> getRequiredExtension() const override {
+    if (getOpCode() == OpAtomicFAddEXT)
+      return ExtensionID::SPV_EXT_shader_atomic_float_add;
+    return {};
+  }
```

Comment:
As a future improvement I think it makes sense to create a map `OpCode` -> `ExtensionID`.

Reply:
To clarify, is your suggestion to map all of the `OpCode`s requiring some extension to the corresponding `ExtensionID`s? Unless there are cases where the extension's requirement is conditional, I also believe this could be a great improvement. For one, this could drop the need for having `getRequiredExtension()` as `virtual`.

As for this particular bit, I've actually had a slightly different refactoring idea, namely to specialize functions for the macro-generated `SPIRVAtomic*` classes (since the `RequiredCapability` specialization also wants for some improvement). This would probably include some rework of the `__SPIRV_OP` macro - I haven't included such attempts into this PR, but creating a follow-up could be a plan.